modules/scheduler-impl/src/main/java/org/opencastproject/scheduler/impl/persistence/SchedulerServiceDatabaseImpl.java

```diff
@@ -169,40 +159,79 @@ public Date getLastModified(String agentId) throws NotFoundException, SchedulerS
-      return dto.getId();
-    } catch (NoResultException e) {
-      throw new NotFoundException("Transaction with source " + source + " does not exist");
-    } catch (Exception e) {
-      logger.error("Could not retrieve id for transaction with source '{}': {}", source, getStackTrace(e));
-      throw new SchedulerServiceDatabaseException(e);
-    } finally {
-      if (em != null)
-        em.close();
-    }
-  }
-
-  @Override
-  public String getTransactionSource(String id) throws NotFoundException, SchedulerServiceDatabaseException {
+  public void storeEvent(String mediapackageId, String organizationId, Opt<String> captureAgentId, Opt<Date> start,
+          Opt<Date> end, Opt<String> source, Opt<String> recordingState, Opt<Long> recordingLastHeard,
+          Opt<String> reviewStatus, Opt<Date> reviewDate, Opt<String> presenters, Opt<Boolean> optOut,
+          Opt<String> lastModifiedOrigin, Opt<Date> lastModifiedDate, Opt<String> checksum,
+          Opt<Map<String, String>> workflowProperties, Opt<Map<String, String>> caProperties
+  ) throws SchedulerServiceDatabaseException {
     EntityManager em = null;
+    EntityTransaction tx = null;
     try {
       em = emf.createEntityManager();
-      Opt<TransactionDto> entity = getTransactionDto(id, em);
-      if (entity.isNone())
-        throw new NotFoundException("Transaction with ID " + id + " does not exist");
+      tx = em.getTransaction();
+      tx.begin();
+      Opt<ExtendedEventDto> entityOpt = getExtendedEventDto(mediapackageId, organizationId, em);
+      ExtendedEventDto entity = entityOpt.getOr(new ExtendedEventDto());
+      entity.setMediaPackageId(mediapackageId);
+      entity.setOrganization(organizationId);
+      if (captureAgentId.isSome()) {
+        entity.setCaptureAgentId(captureAgentId.get());
+      }
+      if (start.isSome()) {
+        entity.setStartDate(start.get());
+      }
+      if (end.isSome()) {
+        entity.setEndDate(end.get());
+      }
+      if (source.isSome()) {
+        entity.setSource(source.get());
+      }
+      if (recordingState.isSome()) {
+        entity.setRecordingState(recordingState.get());
+      }
+      if (recordingLastHeard.isSome()) {
+        entity.setRecordingLastHeard(recordingLastHeard.get());
+      }
+      if (reviewStatus.isSome()) {
+        entity.setReviewStatus(reviewStatus.get());
+      }
+      if (reviewDate.isSome()) {
+        entity.setReviewDate(reviewDate.get());
+      }
+      if (presenters.isSome()) {
+        entity.setPresenters(presenters.get());
+      }
+      if (optOut.isSome()) {
+        entity.setOptOut(optOut.get());
+      }
+      if (lastModifiedOrigin.isSome()) {
+        entity.setLastModifiedOrigin(lastModifiedOrigin.get());
+      }
+      if (lastModifiedDate.isSome()) {
+        entity.setLastModifiedDate(lastModifiedDate.get());
+      }
+      if (checksum.isSome()) {
+        entity.setChecksum(checksum.get());
+      }
+      Gson gson = new Gson();
```

Comment:
Maybe use a class level var? We don't need to initialize Gson every time, do we?